Analyzer/plugins/Analyzer.cc

```diff
@@ -3542,6 +3548,42 @@ void Analyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
               // Pixel corrections
               float pixelScaling = GetSFPixel(detid.subdetId(), detid, year, run_number);
               chargeForIndxH *= pixelScaling;
+
+
+              // Taking the pixel cluster
+              auto const* pixelCluster =  dedxHits->pixelCluster(h);
+              if (pixelCluster == nullptr)  continue;
+              // Check on which geometry unit the hit is
+              const GeomDetUnit& geomDet = *tkGeometry->idToDetUnit(detid);
+              // Get the local vector for the track direction
+              LocalVector lv = geomDet.toLocal(GlobalVector(track->px(), track->py(), track->pz()));
+              // Re-run the CPE on this cluster with the lv above
+              // getParameters will return std::tuple<LocalPoint, LocalError, SiPixelRecHitQuality::QualWordType>;
+              // from this we pick the 2nd, the QualWordType
+              auto reCPE = std::get<2>(pixelCPE->getParameters(*pixelCluster, geomDet, LocalTrajectoryParameters(dedxHits->pos(h), lv, track->charge())));
+              // extract probQ and probXY from this
+              //float probQ = SiPixelRecHitQuality::thePacking.probabilityQ(reCPE);
+              // To measure how often the CPE fails
+              bool cpeHasFailed = false;
+              if (!SiPixelRecHitQuality::thePacking.hasFilledProb(reCPE)) {
+                  cpeHasFailed = true;
+              }
+              if (cpeHasFailed) continue;
+
+             //  if (probQ <= 0.0 || probQ >= 1.f) probQ = 1.f;
+
+              bool isOnEdge = SiPixelRecHitQuality::thePacking.isOnEdge(reCPE);
+              bool hasBadPixels = SiPixelRecHitQuality::thePacking.hasBadPixels(reCPE);
+              bool spansTwoROCs = SiPixelRecHitQuality::thePacking.spansTwoROCs(reCPE);
+              bool specInCPE = (isOnEdge || hasBadPixels || spansTwoROCs) ? true : false;
+
+             //  bool isBPIXL1=false;
+             //  int numLayers = tkGeometry->numberOfLayers(PixelSubdetector::PixelBarrel);
+             //  if ((numLayers == 4) && ((detid.subdetId() == PixelSubdetector::PixelBarrel) && (tTopo->pxbLayer(detid) == 1))) isBPIXL1=true;
+
```

Comment:
is there a reason why you commented this and not removed it?

Reply:
I decide to keep the commented lines in case we decide at some point to also remove BPix one in GiTemplates...